deployments/kubehound/notebook/LowHangingFruit-ContainerEscape.ipynb

```diff
@@ -0,0 +1,492 @@
+    "    .where(\n",
+    "        repeat(\n",
+    "          outE().inV().simplePath()  // Building the path\n",
+    "        ).until(\n",
+    "            has(label, \"Node\")       // Stop when meeting a critical asset\n",
+    "            .or().loops().is(10)     // Stop after 4 iteration\n",
+    "        ).has(label, \"Node\")         // Keep only path ending with a critical asset\n",
+    "        .limit(1)\n",
+    "    )\n",
+    "    .by(\"namespace\")\n",
+    "    .by(\"app\")\n",
+    "    .valueMap(\"namespace\",\"app\")\n",
+    "    .limit(1000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "frozen": false,
+    "tags": [
+     "unsafe_output"
+    ]
+   },
+   "source": [
+    "## Manual investigation for each app/namespace\n",
+    "\n",
+    "From the above list, you can iterate manual investigation by scoping by each vulnerable `app`/`namespace`. To process the investigation, just copy/paste the name of the vulnerable app (replace `VULNERABLE_APP` by the targetted app).\n",
+    "\n",
+    "### Listing all attack paths from a particular app\n",
+    "\n",
+    "The following gremlin request will **list all container escapes**. We add a limit(1000) to avoid having huge graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {
+    "frozen": false,
+    "scrolled": true,
+    "tags": [
+     "safe_output"
+    ]
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "9ca4a7615c304b8cb2a41cc68e5fd829",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Tab(children=(Output(layout=Layout(max_height='600px', max_width='940px', overflow='scroll')), Force(network=<…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%%gremlin -d class -g critical -le 50 -p inv,oute\n",
+    "\n",
+    "kh.containers()\n",
+    "    .has(\"runID\", graph.variables().get('runID_yourid').get())\n",
+    "    .has(\"app\",\"VULNERABLE_APP\")\n",
+    "    .repeat(\n",
+    "      outE().inV().simplePath()  // Building the path\n",
+    "    ).until(\n",
+    "        has(label, \"Node\")    // Stop when meeting a critical asset\n",
+    "        .or().loops().is(10)      // Stop after 4 iteration\n",
+    "    ).has(label, \"Node\")       // Keep only path ending with a critical asset\n",
+    "    .path().by(elementMap())\n",
+    "    .limit(1000)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "frozen": false,
+    "tags": [
+     "unsafe_output"
+    ]
+   },
+   "source": [
+    "The last view can be quite overwhelming. Also the last view can be capped with the limit(1000), so we dont have an exhaustive view. Increasing the limit will not solve the issue as, it will become non human readable.  \n",
+    "\n",
+    "### Listing all attack path deduplicated by app from a particular app \n",
```

Comment:
There is no by because we already strip all the information we did not want. So here it will just dedup all the path (there is no vertex_ID, no name, ...).